includes/payment-methods/class-upe-payment-method.php

```diff
@@ -114,6 +122,21 @@ public function is_enabled_at_checkout() {
 		if ( $this->is_subscription_item_in_cart() || $this->is_changing_payment_method_for_subscription() ) {
 			return $this->is_reusable();
 		}
+
+		// This part ensures that when payment limits for the currency declared, those will be respected (e.g. BNPLs).
+		if ( [] !== $this->limits_per_currency ) {
+			$currency = get_woocommerce_currency();
+			// If the currency limits are not defined, we allow the PM for now (gateway has similar validation for limits).
+			// Additionally, we don't engage with limits verification in no-checkout context (cart is not available or empty).
+			if ( isset( $this->limits_per_currency[ $currency ], WC()->cart ) ) {
+				$amount = WC_Payments_Utils::prepare_amount( WC()->cart->get_total( '' ), $currency );
+				if ( $amount > 0 ) {
+					$range = $this->limits_per_currency[ $currency ];
+					return $amount >= $range['min'] && $amount <= $range['max'];
```

Comment:
What if we want to have a minimum or a maximum only? In general, such flexibility would allow us to have a client-side checks for any payment method <-> currency combination.

Reply:
Hm, that makes sense. I'll craft a new PR shortly.